Add GH workflow for release PR & checklist

tomalec opened the pull request:
### Changes proposed in this Pull Request:

Add a manual GH workflow that takes versions, then starts a branch and creates a PR with the checklist.
The aim is to automate and unify the release process across our repos.

### Screenshots:

![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/685aaa36-0165-4164-8346-cda3a6ba8b69)




### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

Unfortunately, you cannot test a manually triggered workflow, until it's merged to the default branch :( 
But this PR follows,
 https://github.com/woocommerce/automatewoo/pull/1435/ and https://github.com/woocommerce/google-listings-and-ads/pull/1982


### Additional details:


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Dev - Add release preparation GH workflow.

tomalec committed:
Add GH workflow for release PR & checklist

eason9487 commented on .github/workflows/prepare-release.yml:
This extension is published to wordpress.org as well, so [a step to remove older changelog entries from `readme.txt`](https://github.com/woocommerce/google-listings-and-ads/blob/2.4.11/.github/workflows/prepare-release.yml#L40-L41) could be added to the `pre-steps` parameter.

eason9487 reviewed (approved):
LGTM.

tomalec committed:
Add a release step to trim changelog in the readme.txt

Addresses https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/285#discussion_r1266149285
